src/components/AuthoritiesLookup/AuthoritiesLookup.js

```diff
@@ -0,0 +1,184 @@
+import {
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import { useIntl } from 'react-intl';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { Pane } from '@folio/stripes/components';
+import { AppIcon } from '@folio/stripes-core';
+import {
+  AuthoritiesSearchContext,
+  AuthoritiesSearchPane,
+  AuthorityShape,
+  navigationSegments,
+  SearchResultsList,
+  SelectedAuthorityRecordContext,
+} from '@folio/stripes-authority-components';
+
+import MarcAuthorityView from '../MarcAuthorityView';
+import {
+  columnWidths,
+  MAIN_PANE_HEIGHT,
+  PAGE_SIZE,
+} from '../../constants';
+import css from './AuthoritiesLookup.css';
+
+const propTypes = {
+  authorities: PropTypes.arrayOf(AuthorityShape).isRequired,
+  hasFilters: PropTypes.bool.isRequired,
+  hasNextPage: PropTypes.bool,
+  hasPrevPage: PropTypes.bool,
+  hidePageIndices: PropTypes.bool,
+  isLoaded: PropTypes.bool.isRequired,
+  isLoading: PropTypes.bool.isRequired,
+  onNeedMoreData: PropTypes.func.isRequired,
+  onSubmitSearch: PropTypes.func.isRequired,
+  query: PropTypes.string.isRequired,
+  searchQuery: PropTypes.string.isRequired,
+  totalRecords: PropTypes.number.isRequired,
+};
+
+const AuthoritiesLookup = ({
+  authorities,
+  totalRecords,
+  searchQuery,
+  query,
+  isLoaded,
+  isLoading,
+  hasFilters,
+  hasNextPage,
+  hasPrevPage,
+  hidePageIndices,
+  onNeedMoreData,
+  onSubmitSearch,
+}) => {
+  const intl = useIntl();
+  const [isFilterPaneVisible, setIsFilterPaneVisible] = useState(true);
+  const [showDetailView, setShowDetailView] = useState(false);
+
+  const { navigationSegmentValue } = useContext(AuthoritiesSearchContext);
+  const [, setSelectedAuthorityRecordContext] = useContext(SelectedAuthorityRecordContext);
+
+  const toggleFilterPane = () => setIsFilterPaneVisible(!isFilterPaneVisible);
+
+  const closeDetailView = () => {
+    setShowDetailView(false);
+    setSelectedAuthorityRecordContext(null);
+  };
+
+  useEffect(() => {
+    if (totalRecords !== 1) {
```

Comment:
This code is duplicated in `ui-marc-authorities` with only difference in _how_ the record is opened
Let's move this code to `SearchResultsList` and inside `if (isDetailViewNeedsToBeOpened)` call a prop `openRecord`. Then we can pass whatever logic is used to open the record as a prop and have minimal code duplication

Reply:
Thanks for the comment. In the _plugin_, `useEffect` will be triggered permanently if we shift it to `SearchResultsList` because the detail view is unmounted each time we close it.

Example (in the plugin):
When detail view is open: click close detail view => detail view is unmounted => `SearchResultsList` is mounted with one record => useEffect is invoked => detail view is mounted again

I moved common functionality of `useEffect` to utils as `getIsDetailViewNeedsToBeOpened`.